### PR TITLE
Add eval step

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This PR includes:

* Minor fix to _utils.dtype_torch_to_numpy
* Add minimal ORTTrainer.eval_step() functionality
* extended unit test to cover eval_step

### CODE REVIEW

File: orttraining/orttraining/python/training/_utils.py [hunk 1]
```diff
@@ -22,7 +38,7 @@ def dtype_torch_to_numpy(torch_dtype):
         # NOTE: numpy doesn't support bfloat16
         return np.float16
     elif torch_dtype == torch.int64 or torch_dtype == torch.long:
-        return np.int64
+        return np.longlong # np.int64 doesn't work!?
```
Comment (rayankrish) [inline orttraining/orttraining/python/training/_utils.py:41]: NIP: should we delete this comment?

File: orttraining/orttraining/python/training/orttrainer.py [hunk 2]
```diff
@@ -156,23 +171,64 @@ def eval_step(self, *input, **kwargs):
...
+                raise RuntimeError("Model is uninitialized. Only ONNX and PyTorch models are supported")
+
+        input_desc = self.model_desc.inputs[0:len(sample_input)]
+        output_desc = self.model_desc.outputs
+
+        if not isinstance(sample_input, (list, tuple)):
+            sample_input = (sample_input,)
+
+        run_options = ort.RunOptions()
+        run_options.only_execute_path_to_fetches = True
```
Comment (rayankrish) [inline orttraining/orttraining/python/training/orttrainer.py:191]: should this be True for this PR where we don't have fetches yet?

File: orttraining/orttraining/python/training/orttrainer.py [hunk 3]
```diff
@@ -156,23 +171,64 @@ def eval_step(self, *input, **kwargs):
...
+        run_options.training_mode = False
+        session_run_results = self._training_session_run_helper(False,
+                                                                sample_input,
+                                                                input_desc,
+                                                                output_desc,
+                                                                self.options.device.id,
```
Comment (rayankrish) [inline orttraining/orttraining/python/training/orttrainer.py:197]: could we access the device id directly in the method without passing as a parameter?

File: orttraining/orttraining/python/training/_utils.py (hunk 1 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/_utils.py:41]: I believe this comment will save from people to try to use np.int64 to map a torch.int64.

File: orttraining/orttraining/python/training/orttrainer.py (hunk 2 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:191]: We are not exposing fetches as a part of the public API, but we can use it internally

File: orttraining/orttraining/python/training/orttrainer.py (hunk 3 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:197]: yes, we could. good catch!

**Pull request merged**